modules/utils/validation.js

```diff
@@ -400,7 +407,8 @@ export const getNewValueForFieldOp = function (config, oldConfig = null, current
 					const [validateError, fixedValue] = validateValue(config, newField, newField, newOperator, v, vType, vSrc, canFix, isEndValue);
 					const isValid = !validateError;
 					if (!isValidSrc || !isValid) {
-							canReuseValue = false;
+							if (v !== undefined)
```

Comment:
This now allows to preserve value after change op from `==` to `between` and vice versa